src/components/NavBar/NavBar.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import classes from './NavBar.module.css';
+import { RxCross1 } from 'react-icons/rx';
+import { IconButton } from '@mui/material';
+import { Link } from 'react-router-dom';
+
+const NavBar = ({ isNavBarOpen, toggleNavBarStatus }) => {
+  const [isVisible, setIsVisible] = useState(isNavBarOpen);
+
+  const handleTransitionEnd = () => {
+    !isNavBarOpen && setIsVisible(false);
+  };
+
+  useEffect(() => {
+    isNavBarOpen && setIsVisible(true);
+  }, [isNavBarOpen]);
+
+  const navbarClasses = [classes.navbar];
+
+  isNavBarOpen
+  ? navbarClasses.push(classes['slideIn'])
+  : navbarClasses.push(classes['slideOut']);
```

Comment:
better to use useEffect here, because you never know when will isNavBarOpen update
```
useEffect(() => {
  isNavBarOpen
  ? navbarClasses.push(classes['slideIn'])
  : navbarClasses.push(classes['slideOut']);
}, [isNavBarOpen])
```